### ISSUE TITLE
Add separate modules for metrics and tracing wrappers

### ISSUE THREAD

**Issue by brancz** [thanos-io/objstore#2]
The specific libraries and mechanisms used for metrics and tracing can be quite opinionated and a library meant as an abstraction for object storage providers should not force a dependency of one or the other system onto downstream users. As such I think that the:

* prometheus wrapper (already exists)
* opentracing wrapper (already exists)
* opentelemetry wrapper (does not exist yet)

Should be served by separate modules.

**Comment by bboreham** [comment]
OpenTracing has been deprecated for nearly a year; why wouldn't you just switch to OpenTelemetry?
Adding abstractions over abstractions seems like an antipattern to me.

There is a [bridge](https://github.com/open-telemetry/opentelemetry-go/tree/main/bridge/opentracing) for anyone who needs OpenTracing.

**Comment by brancz** [comment]
I agree I wouldn’t want yet another abstraction, but for me it was important people choose to adopt the dependency, not us force it onto people.

**Comment by brancz** [comment]
I'm not exactly sure what you're proposing.
